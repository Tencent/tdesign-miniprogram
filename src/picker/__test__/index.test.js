import simulate from 'miniprogram-simulate';
import path from 'path';

describe('picker', () => {
  const id = load(path.resolve(__dirname, `./index`));
  const handler = (dom) => {
    const trigger = (action, x, y) =>
      dom.dispatchEvent(`touch${action}`, {
        touches: [{ x, y }],
      });
    return {
      start: (x, y) => trigger('start', x, y),
      move: (x, y) => trigger('move', x, y),
      end: () => trigger('end'),
    };
  };

  it(':base', () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const $city = comp.querySelector('#city');
    comp.setData({ cityVisible: true });

    expect($city.toJSON()).toMatchSnapshot();
  });

  it(':change', async () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const $picker = comp.querySelector('#city');
    const $pickItem = comp.querySelector('#cityItem >>> .t-picker-item__group');
    const pickerHandler = handler($pickItem);

    comp.setData({ cityVisible: true });
    pickerHandler.start(0, 0);
    pickerHandler.move(0, -100);
    pickerHandler.end();
    await simulate.sleep(100);

    const $confirm = $picker.querySelector('.t-picker__confirm');

    $confirm.dispatchEvent('tap');
    await simulate.sleep();

    expect(comp.instance.data.cityValue).toStrictEqual(['深圳']);

    pickerHandler.start(0, 0);
    pickerHandler.move(0, 0);
    pickerHandler.end();
    await simulate.sleep(100);

    $picker.instance.setData({ autoClose: false });
    $confirm.dispatchEvent('tap');
    await simulate.sleep();

    expect(comp.instance.data.cityValue).toStrictEqual(['深圳']);
    expect($picker.instance.data.visible).toBeTruthy();
  });
});
