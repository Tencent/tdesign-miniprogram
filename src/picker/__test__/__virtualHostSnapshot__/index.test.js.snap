// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`picker :base 1`] = `
<wx-view
  ariaModal="{{true}}"
  ariaRole="dialog"
  class="t-popup t-popup--bottom t-fade-enter t-fade-enter-active class t-class"
  style="z-index:11500;"
  bind:transitionend="onTransitionEnd"
>
  <wx-view
    class="t-popup__content t-class-content"
  >
    <wx-view
      class="t-picker t-class"
      slot="content"
      style=""
    >
      <wx-view
        class="t-picker__toolbar"
      >
        <wx-view
          class="t-picker__cancel t-class-cancel"
          bind:tap="onCancel"
        >
          取消
        </wx-view>
        <wx-view
          class="t-picker__title t-class-title"
        >
          
        </wx-view>
        <wx-view
          class="t-picker__confirm t-class-confirm"
          bind:tap="onConfirm"
        >
          确认
        </wx-view>
      </wx-view>
      <wx-view
        class="t-picker__main"
      >
        <wx-view
          class="t-picker-item__group class t-class"
          style=""
          bind:touchcancel="onTouchEnd"
          bind:touchend="onTouchEnd"
          catch:touchmove="onTouchMove"
          bind:touchstart="onTouchStart"
        >
          <wx-view
            class="t-picker-item__wrapper"
            style="transition: transform 0ms cubic-bezier(0.215, 0.61, 0.355, 1); transform: translate3d(0, 0px, 0)"
          >
            <wx-view
              class="t-picker-item__item t-picker-item__item--active"
              data-index="{{0}}"
            >
              
      2021年
    
            </wx-view>
            <wx-view
              class="t-picker-item__item"
              data-index="{{1}}"
            >
              
      2020年
    
            </wx-view>
            <wx-view
              class="t-picker-item__item"
              data-index="{{2}}"
            >
              
      2019年
    
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="t-picker-item__group class t-class"
          style=""
          bind:touchcancel="onTouchEnd"
          bind:touchend="onTouchEnd"
          catch:touchmove="onTouchMove"
          bind:touchstart="onTouchStart"
        >
          <wx-view
            class="t-picker-item__wrapper"
            style="transition: transform 0ms cubic-bezier(0.215, 0.61, 0.355, 1); transform: translate3d(0, 0px, 0)"
          >
            <wx-view
              class="t-picker-item__item t-picker-item__item--active"
              data-index="{{0}}"
            >
              
      春
    
            </wx-view>
            <wx-view
              class="t-picker-item__item"
              data-index="{{1}}"
            >
              
      夏
    
            </wx-view>
            <wx-view
              class="t-picker-item__item"
              data-index="{{2}}"
            >
              
      秋
    
            </wx-view>
            <wx-view
              class="t-picker-item__item"
              data-index="{{3}}"
            >
              
      冬
    
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="t-picker__mask t-picker__mask--top"
        />
        <wx-view
          class="t-picker__mask t-picker__mask--bottom"
        />
        <wx-view
          class="t-picker__indicator"
        />
      </wx-view>
    </wx-view>
    <wx-view
      class="t-popup__close"
      bind:tap="handleClose"
    />
  </wx-view>
</wx-view>
`;
